components/region_cache_memory_engine/src/background.rs

```diff
@@ -451,6 +547,62 @@ impl BackgroundRunnerCore {
         #[cfg(test)]
         flush_epoch();
     }
+
+    fn top_regions_load_evict(&mut self) {
+        if self.range_stats_manager.is_none() {
+            return;
+        }
+        let range_stats_manager = self.range_stats_manager.as_mut().unwrap();
+        if range_stats_manager.checking_top_regions() {
+            return;
+        }
+        range_stats_manager.set_checking_top_regions(true);
+
+        let curr_memory_usage = self.memory_controller.mem_usage();
+        let threshold = self.memory_controller.soft_limit_threshold();
+
+        if curr_memory_usage < threshold {
```

Comment:
@tonyxuqqi @SpadeA-Tang you may want to pay closer attention to this. I am trying to verify this experimentally in my setup, that this approximates doing the right thing.